Packs/FireEyeHX/Integrations/FireEyeHXEventCollector/FireEyeHXEventCollector.py

```diff
@@ -0,0 +1,224 @@
+
+        token = demisto.getIntegrationContext().get('token')
+        if not token:
+            token = self.get_access_token()
+
+        headers = {'X-FeApi-Token': token}
+        try:
+            res = super()._http_request(*args, headers=headers, **kwargs)  # type: ignore[misc]
+        except Exception as e:
+            # in case the token expired - create the token again and make the request.
+            if e.res.status_code == 401:  # type: ignore[attr-defined]
+                token = self.get_access_token()
+                headers = {'X-FeApi-Token': token}
+                res = super()._http_request(*args, headers=headers, **kwargs)  # type: ignore[misc]
+            else:
+                raise
+        return res
+
+    def get_access_token(self) -> str:
+        """
+       Obtains access and refresh token from server.
+       Access token is used and stored in the integration context.
+
+        Returns:
+            str: the access token.
+       """
+
+        token_response = self._http_request('GET', url_suffix='/hx/api/v3/token', auth=(self.username, self.password),
+                                            resp_type='response', ok_codes=[204])
+
+        token = token_response.headers.get('X-FeApi-Token')
+        demisto.setIntegrationContext({'token': token})
+        return token
+
+    def get_events_request(self, limit: str = '100', min_id: str = None, filter_query: str = None,
+                           resolution: str = None):
+        """
+        Get alerts from fireeye
+        """
+
+        params = assign_params(resolution=resolution,
+                               sort='event_at+ascending',
+                               limit=limit,
+                               min_id=min_id,
+                               filterQuery=filter_query)
+
+        return self.http_request(
+            'GET',
+            url_suffix='/hx/api/v3/alerts',
+            ok_codes=[200],
+            params=params
+        )
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client):
+    """
+    Testing we have a valid connection to fireeye.
+    """
+    client.get_events_request()
+    return 'ok'
+
+
+def get_events_command(
+    client: Client,
+    max_fetch: str,
+    first_fetch: str
+) -> Union[str, CommandResults]:
+    """
+    Fetches events from the fireeye and return them to the war-room.
+    """
+
+    events = fetch_events(
+        client=client, max_fetch=max_fetch, first_fetch=first_fetch)
+
+    if not events:
+        return 'No events were found.'
+
+    return CommandResults(
+        readable_output=tableToMarkdown(
+            'FireEye HX events',
+            events,
+            headerTransform=underscoreToCamelCase,
+            removeNull=True
+        ),
+        raw_response=events,
+        outputs=events,
+        outputs_prefix='FireEyeHX.Event'
+    )
+
+
+def fetch_events(
+    client: Client, max_fetch: str, first_fetch: str, resolution: str = None, min_id: str = None
```

Comment:
Better to use list and not List and dict and not Dict because of the [deprecation](https://www.python.org/dev/peps/pep-0585/#implementation).